### PR TITLE
[LLM] Supports GPTQ convert in transfomers-like API, and supports folder outfile for `llm-convert`

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Supports GPTQ convert in transfomers-like API, and supports folder outfile for `llm-convert`

https://github.com/orgs/analytics-zoo/projects/21/views/3?pane=issue&itemId=31133158
https://github.com/orgs/analytics-zoo/projects/21/views/3?pane=issue&itemId=31133527

### 1. User API changes
- `bigdl.llm.llm_convert`: now `outfile` requires folder path as inputs
- `bigdl.llm.transformers.from_pretrained`: add param `model_format`

### 2. Summary of the change 
- Supports GPTQ convert in `bigdl.llm.transformers.from_pretrained`
- Unify `outfile` definition for `bigdl.llm.llm_convert`, now only folder path is valid
- Add docstrings for `bigdl.llm.llm_convert`
- Fix bug in `bigdl.llm.llm_convert`
- Revise related examples

### 3. How to test?
- [x] Local test
- [x] Unit test

### CODE REVIEW

File: python/llm/src/bigdl/llm/convert_model.py [hunk 1]
```diff
@@ -35,6 +36,44 @@ def llm_convert(model,
...
+    :param **kwargs: Supported keyword arguments includes:
+
+           * ``tmp_path``: Valid when ``model_format='pth'``. It refers to the path
+             that stores the intermediate model during the conversion process.
+           * ``tokenizer_path``: Valid when ``model_format='gptq'``. It refers to the path
+             where ``tokenizer.model`` is located (if it is not in the ``model`` directory)
+
+    :return: the path string to the converted lower precision checkpoint.
+    """
     if model_format == "pth":
```
Comment (TheaperDeng) [inline python/llm/src/bigdl/llm/convert_model.py:77]: I am not sure if pth is the best string for hf ckpt...

File: python/llm/src/bigdl/llm/convert_model.py [hunk 2]
```diff
@@ -48,11 +87,23 @@ def llm_convert(model,
         invalidInputError(model_family == "llama" and outtype == 'int4',
                           "Convert GPTQ models should always "
                           "specify `--model-family llama --dtype int4` in the command line.")
+        invalidInputError(os.path.isdir(outfile),
+                          "The output_path {} was not a directory".format(outfile))
```
Comment (TheaperDeng) [inline python/llm/src/bigdl/llm/convert_model.py:91]: is not

**Review by TheaperDeng — APPROVED** [review]

File: python/llm/src/bigdl/llm/convert_model.py (hunk 1 above)
Comment (Oscilloscope98) [reply python/llm/src/bigdl/llm/convert_model.py:78]: I guess pth refers to **P**y**T**orc**h**

File: python/llm/src/bigdl/llm/convert_model.py (hunk 2 above)
Comment (Oscilloscope98) [reply python/llm/src/bigdl/llm/convert_model.py]: Has fixed

**Review by leonardozcm — APPROVED** [review]

**Pull request merged**